functions/src/delete_shared_schedule.ts

```diff
@@ -61,38 +71,38 @@ export const deleteInvitationFromFriend = functions.https.onRequest(
         }
 
         // Get user UID from the decoded token
-        const friendId = decodedToken.uid;
+        const requesterId = decodedToken.uid;
 
-        const senderRes = await schedulesCollection.doc(senderId).get();
-        const senderData: Version3ScheduleData | undefined =
-          senderRes.data() as Version3ScheduleData | undefined;
+        const senderId = owner ? requesterId : otherUserId;
+        const friendId = owner ? otherUserId : requesterId;
 
-        const flag = !!(
-          senderData &&
-          senderData.terms &&
-          senderData.terms[term].versions
-        );
+        const existingInvites = await invitesCollection
+          .where("sender", "==", senderId)
+          .where("friend", "==", friendId)
+          .where("term", "==", term)
+          // .where("version", "==", version)
+          .get();
+
+        const scheduleResponse = await schedulesCollection.doc(senderId).get();
+        const scheduleData: Version3ScheduleData | undefined =
+          scheduleResponse.data() as Version3ScheduleData | undefined;
+
+        const friendData = (await friendsCollection.doc(friendId).get()).data();
+        const accessibleSchedules =
+          friendData?.terms?.[term]?.accessibleSchedules;
 
         // find and delete existing invites for the same sender, friend, term, and version
         // also deletes friend invites that show up on the sender's invitation modal
         await Promise.allSettled(
           versions.map(async (version) => {
             try {
-              const existingInvites = await invitesCollection
-                .where("sender", "==", senderId)
-                .where("friend", "==", friendId)
-                .where("term", "==", term)
-                // .where("version", "==", version)
-                .get();
               const batch = firestore.batch();
               existingInvites.forEach((doc) => {
                 if (doc.get("link")) {
                   return;
                 }
                 const currVersions: string[] = doc.get("versions");
-                const newVersions = currVersions.filter(
-                  (v) => !versions.includes(v)
-                );
+                const newVersions = currVersions.filter((v) => v !== version);
                 if (newVersions.length === 0) {
                   batch.delete(doc.ref);
                 } else if (newVersions.length !== currVersions.length) {
```

Comment:
make this an else statement and add comment for when this block of code will be hit